nucypher/network/server.py

```diff
@@ -113,56 +113,57 @@ def _make_rest_app(datastore: Datastore, this_node, domain: str, log: Logger) ->
 
     @rest_app.route("/public_information")
     def public_information():
-        """
-        REST endpoint for public keys and address.
-        """
-        response = Response(
-            response=bytes(this_node),
-            mimetype='application/octet-stream')
-
+        """REST endpoint for public keys and address."""
+        response = Response(response=bytes(this_node), mimetype='application/octet-stream')
         return response
 
-    @rest_app.route("/ping", methods=['POST'])
+    @rest_app.route("/ping", methods=['GET', 'POST'])
     def ping():
         """
-        Asks this node: "Can you access my public information endpoint"?
+        GET: Asks this node: "What is my IP address?"
+        POST: Asks this node: "Can you access my public information endpoint?"
         """
 
-        try:
-            requesting_ursula = Ursula.from_bytes(request.data)
-            requesting_ursula.mature()
-        except ValueError:  # (ValueError)
-            return Response({'error': 'Invalid Ursula'}, status=400)
-        else:
-            initiator_address, initiator_port = tuple(requesting_ursula.rest_interface)
-
-        # Compare requester and posted Ursula information
-        request_address = request.environ['REMOTE_ADDR']
-        if request_address != initiator_address:
-            return Response({'error': 'Suspicious origin address'}, status=400)
+        if request.method == 'GET':
+            requester_ip_address = request.environ['REMOTE_ADDR']
+            return Response(requester_ip_address, status=200)
 
-        #
-        # Make a Sandwich
-        #
+        elif request.method == 'POST':
+            try:
+                requesting_ursula = Ursula.from_bytes(request.data)
+                requesting_ursula.mature()
+            except ValueError:
+                return Response({'error': 'Invalid Ursula'}, status=400)
+            else:
+                initiator_address, initiator_port = tuple(requesting_ursula.rest_interface)
+
+            # Compare requester and posted Ursula information
+            request_address = request.environ['REMOTE_ADDR']
+            if request_address != initiator_address:
+                return Response({'error': 'Origin address mismatch'}, status=400)
```

Comment:
Would be useful to include `request_address` and `initiator_address` in the error.

Reply:
:heavy_check_mark: - Not introduced in this PR but a good suggestion nonetheless.